### PR TITLE
Add ability to control the target for a button

### LINKED ISSUE #237: Add ability to set target on buttons
image
If I use this type, I am not able to set the link to be opened in another tab/window. 

I believe this would be an easy fix, are you open for a community contribution on this?

### PR DESCRIPTION
**Pull request by gabiudrescu**
Fix #237

### CODE REVIEW

File: src/Form/Type/UiElement/ButtonLinkType.php [hunk 1]
```diff
@@ -67,6 +68,12 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                 'label' => 'monsieurbiz_richeditor_plugin.ui_element.monsieurbiz.button.field.link',
                 'constraints' => $constraintsLink,
             ])
+            ->add('target', ChoiceType::class, [
+                'choices'  => [
+                    'Open in the same frame as it was clicked (this is default)' => '_self',
+                    'Open in a new window or tab' => '_blank',
```
Comment (maximehuran) [inline src/Form/Type/UiElement/ButtonLinkType.php:74]: Can you translate the labels ?

File: src/Resources/views/Admin/UiElement/button.html.twig [hunk 2]
```diff
@@ -4,9 +4,11 @@
...
         label
         link
+        target
         align
 #}
 {% set align = element.align is defined and element.align is not empty ? element.align : 'center' %}
+{% set target = element.target is defined and element.target is not empty ? element.target : '_self' %}
 <p style="text-align: {{align}};">
-    <a href="{{ element.link }}" class="ui primary button massive">{{ element.label }}</a>
+    <a href="{{ element.link }}" target="{{ element.target }}" rel="noopener noreferrer" class="ui primary button massive">{{ element.label }}</a>
```
Comment (maximehuran) [inline src/Resources/views/Admin/UiElement/button.html.twig:13]: The `rel="noopener noreferrer"` must be added only if target is blank

You can compare if blank with the constant defined before with Twig `constant` method

File: src/Form/Type/UiElement/ButtonLinkType.php (hunk 1 above)
Comment (maximehuran) [reply src/Form/Type/UiElement/ButtonLinkType.php:74]: Also can you use constants for `_self` and `_blank` ?

**Review by maximehuran — CHANGES_REQUESTED** [review]

File: src/Form/Type/UiElement/ButtonLinkType.php (hunk 1 above)
Comment (gabiudrescu) [reply src/Form/Type/UiElement/ButtonLinkType.php]: doh! for sure. now that you point it out, sounds silly of me not to do this by default. thanks.

File: src/Resources/views/Admin/UiElement/button.html.twig (hunk 2 above)
Comment (gabiudrescu) [reply src/Resources/views/Admin/UiElement/button.html.twig]: I've submitted another commit with a potential fix. let me know what you think.

Comment (gabiudrescu) [reply src/Resources/views/Admin/UiElement/button.html.twig]: nah, I still have to tinker around a bit to make it work. will do this tomorrow.

**Review by maximehuran — APPROVED** [review]

**Comment by maximehuran** [comment]
Thank you @gabiudrescu 🚀

**Pull request merged**